providers/shared/components/apigateway/id.ftl

```diff
@@ -93,24 +93,25 @@ object.
             {
                 "Names" : "EndpointType",
                 "Types" : STRING_TYPE,
-                "Values" : ["EDGE", "REGIONAL"],
+                "Values" : ["EDGE", "REGIONAL", "PRIVATE"],
                 "Default" : "EDGE"
             },
             {
-                "Names" : "IPAddressGroups",
+                "Names" : ["TrafficSources", "IPAddressGroups", "VPCEndpoints"],
```

Comment:
How do I define VPC endpoints? based on their aws arn/id? Given IP address groups work off the abstraction within hamlet we probably either need to say how to use it or figure out what the abstraction should be in this case?

Reply:
It is just a DNS name so a string. I played with having a separate entry for them but all the policy logic works pretty well the same whether it is IP or VPC endpoint or overall it kind of felt like the same thing. 

You are correct that the abstraction is a bit different, which is why I allowed for the use of the explicit name depending on what the situation was.

Do you think it deserves its own specific config attribute? Happy to switch to that if you think we should. Technically you could IP whitelist as well as endpoint whitelist, but given the way the private APIs work, I can't see much call for that.

Reply:
Reverted this change and now using inbound links from external service to define consuming vpc endpoints